src/execution_plan/ops/op_update.c

```diff
@@ -143,6 +143,10 @@ void _UpdateSchemas(const OpUpdate *op) {
         LabelStoreType t = (ge->t == N_ENTITY) ? STORE_NODE : STORE_EDGE;
 
         LabelStore *store = LabelStore_Get(op->ctx, t, op->graphName, l);
+        // TODO if this label does not exist, we will not make any updates.
+        // Verify that this behavior matches the Cypher specification.
+        if (!store) return;
```

Comment:
Once we'll enforce label matching while traversing then store is guaranteed to exists. at the moment it's possible that while a specified label doesn't exists we'll still be getting matches, obviously that doesn't make sense, this should be fixed once we'll refactor our execution plan and enforce label matching.